modules/database/src/std/rec/subRecord.c

```diff
@@ -162,7 +162,7 @@ static long process(struct dbCommon *pcommon)
     recGblFwdLink(prec);
     prec->pact = FALSE;
 
-    return 0;
+    return status;
```

Comment:
This may be the most controversial change in this PR, although I think it's likely that we'll be fine with it. To document what I'm seeing: Some record types (e.g. fanoutRecord.c) have always returned `0` from their process() routine; some (e.g. aoRecord.c) have always returned `status`; some (e.g. calcRecord.c) had their `return status` replaced by `return 0` (2b6886385ba0546be3c3eade8079effbf2986489 in 1990), and others (e.g. waveformRecord.c much more recently in commit 23d917677221fc9d24421aa02dc63d2767c563e0) went the other way.

The rules have never been properly written down, the example of `process()` given [in the AppDevGuide](https://epics.anl.gov/base/R3-16/2-docs/AppDevGuide/RecordSupport.html#x12-49000011.4.3) (and I suspect most I/O record types) returns `0` from the first pass of an asynchronous device processing but `status` from the second or from a synchronous device. I don't see anything explaining how the return value is used, IIRC it's only fairly recently that it was actually propagated out of the parent `dbProcess()` routine.